### PR TITLE
OSDOCS-1824: Adding new book for ROSA Performance scaling and limits.

### PR DESCRIPTION
**Pull request by jonquilwilliams**
JIRA: https://issues.redhat.com/browse/OSDOCS-1824

Preview: https://deploy-preview-35045--osdocs.netlify.app/openshift-rosa/latest/rosa_planning/rosa-limits-scalability.html

Peer reviewer: Please merge to `dedicated-4` branch. No milestones or CP. Thanks.

### CODE REVIEW

File: modules/rosa-planning-considerations.adoc [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+The sizing of the control plane and infrastructure nodes is dynamically calculated during the installation process, based on the number of worker nodes. If the number of workers is changed after the installation, control plane and infra nodes will not be resized.
+
+The following table lists the size of control plane and infrastructure nodes that are assigned during installation.
+
+[options="header",cols="3*"]
+|===
+| Number of worker nodes |Control plane size |Infrastructure node size
+
+|1 to 25
+|m5.xlarge
```
Comment (igoihman) [inline modules/rosa-planning-considerations.adoc:22]: m5.2xlarge

File: modules/rosa-planning-considerations.adoc [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+|r5.xlarge
+
+|26 to 100
+|m5.2xlarge
```
Comment (igoihman) [inline modules/rosa-planning-considerations.adoc:26]: m5.4xlarge

File: modules/rosa-planning-considerations.adoc [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+|r5.2xlarge
+
+|100 to 180 ^[1]^
+|m5.4xlarge
```
Comment (igoihman) [inline modules/rosa-planning-considerations.adoc:30]: m5.8xlarge

**Review by igoihman — COMMENTED** [review]
The master instance type sizing has been doubled up recently - I added the updated sizes.

File: modules/rosa-planning-considerations.adoc [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+// rosa_planning/rosa-planning-environment.adoc
+
+[id="initial-planning-considerations_{context}"]
+= Initial planning considerations
+
+Consider the following tested object maximums when you plan your {product-title} cluster.
+
+These guidelines are based on a cluster of 102 workers in a multi-availability zone configuration. For smaller clusters, the maximums are lower.
+
+The sizing of the control plane and infrastructure nodes is dynamically calculated during the installation process, based on the number of worker nodes. If the number of workers is changed after the installation, control plane and infra nodes will not be resized.
```
Comment (yuwang-RH) [inline modules/rosa-planning-considerations.adoc:13]: @jonquilwilliams The control plane and infra nodes should be resized based on the compute node even after the installation. See https://issues.redhat.com/browse/SDA-2102 and https://issues.redhat.com/browse/SDA-3907

File: modules/rosa-planning-environment-application-reqs.adoc [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+       - name: ENV_VALUE
+         description: Value to use for environment variables
+         generate: expression
+         from: "[A-Za-z0-9]{255}"
+         required: false
+       labels:
+         template: deploymentConfigTemplate
+----
+
+The number of application pods that can run in a namespace is dependent on the number of services and the length of the service name when the environment variables are used for service discovery. `ARG_MAX` on the system defines the maximum argument length for a new process and it is set to `2097152 KiB` by default. The Kubelet injects environment variables in to each pod scheduled to run in the namespace including:
```
Comment (mikemckiernan) [inline modules/rosa-planning-environment-application-reqs.adoc:160]: I don't see a reason to enclose `2097152 KiB` in backticks.  If there's a guideline for that, please share.

I'm fairly confident that kublet is just lowercase.  I believe I can find some precedent in the Nodes section of the doc.

File: modules/rosa-planning-environment-cluster-max.adoc [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+[id="planning-environment-cluster-maximums_{context}"]
+= Planning your environment based on tested cluster maximums
+
+This document describes how to plan your {product-title} environment based on the tested cluster maximums.
+
+Oversubscribing the physical resources on a node affects resource guarantees the Kubernetes scheduler makes during pod placement. Learn what measures you can take to avoid memory swapping.
+
+Some of the tested maximums are stretched only in a single dimension. They will vary when many objects are running on the cluster.
+
+The numbers noted in this documentation are based on Red Hat’s test methodology, setup, configuration, and tunings. These numbers can vary based on your own individual setup and environments.
```
Comment (mikemckiernan) [inline modules/rosa-planning-environment-cluster-max.adoc:15]: FWIW, I take the IBM stance on not using the possessive form of a company (around p67 in the ISG).  So, maybe "...on Red Hat testing methodology,..."  Up to you.

File: rosa_planning/rosa-planning-limits.adoc [hunk 7]
```diff
@@ -0,0 +1,11 @@
+include::modules/attributes-openshift-dedicated.adoc[]
+
+[id="rosa-planning-limits-and-scalability"]
+= Planning limits and scalability
```
Comment (mikemckiernan) [inline rosa_planning/rosa-planning-limits.adoc:4]: Friendly reminder that this title doesn't match the value in the topic map file.

**Review by mikemckiernan — COMMENTED** [review]
Please let me know if there's anything I can clarify.

**Pull request merged**